custom_components/tuya_local/devices/hotshot2000_heater.yaml

```diff
@@ -0,0 +1,72 @@
+name: HOT SHOT 2000 WIFI
+products:
+  - id: lgn0xaegjbodrjax
+    name: EUROM Hot-Shot 2000 WiFi
+primary_entity:
+  entity: climate
+  dps:
+    - id: 1
+      type: boolean
+      name: hvac_mode
+      mapping:
+        - dps_val: false
+          value: "off"
+        - dps_val: true
+          constraint: work_mode
+          conditions:
+            - dps_val: level1
+              value: fan_only
+            - dps_val: level2
+              value: heat
+            - dps_val: level3
+              value: heat
+    - id: 2
+      type: integer
+      name: temperature
+      range:
+        min: 5
+        max: 37
+    - id: 3
+      type: integer
+      name: current_temperature
+    - id: 5
+      name: work_mode
+      type: string
+      hidden: true
+    - id: 6
+      name: preset_mode
+      type: boolean
+      mapping:
+        - dps_val: false
+          constraint: work_mode
+          conditions:
+            - dps_val: "level2"
+              value: comfort
+            - dps_val: "level3"
+              value: boost
+        - dps_val: true
+          value: eco
+    - id: 8
+      type: boolean
+      name: swing_mode
+      mapping:
+        - dps_val: false
+          value: "off"
+        - dps_val: true
+          value: "on"
+secondary_entities:
+  - entity: switch
+    name: Light
+    category: diagnostic
+    dps:
+      - id: 10
+        type: boolean
+        name: switch
+  - entity: sensor
+    name: Remaining Time
```

Comment:
Remaining time, but no Timer to go with it?

If the timer is set by setting this same dp, then it should be a number entity rather than a sensor.

Reply:
Unfortunately it is not possible to set this, it is read only. I tried "number" instead of "sensor" but when I set a value in the UI the device resets it back. It is a 9 hours timer and it starts when you turn on the device.

Reply:
If it is a fixed timer, that is always active when the device is on, then it makes sense without the corresponding Timer select or number  entity.